src/utils/request.ts

```diff
@@ -49,24 +49,44 @@ export function request(options: IRequestOptions): Promise<IRequestContext> {
 			opts.auth = options.user + ':' + options.password;
 		}
 
-		req = protocol.request(opts, (res: http.ClientResponse) => {
-			const followRedirects = isNumber(options.followRedirects) ? options.followRedirects : 3;
+		if (protocol === "https")
```

Comment:
I do not like the fact that this change causes of duplicate code. I find duplicate code causes issues with maintainability since now bug fixes need to be made in multiple area. If we like this if/else as opposed to what was before I recommend separating the call back into it's one function